src/ast/passes/codegen_llvm.cpp

```diff
@@ -466,8 +466,21 @@ void CodegenLLVM::visit(Builtin &builtin)
     expr_ = b_.CreateLoad(b_.GetType(builtin.type), dst);
     b_.CreateLifetimeEnd(dst);
   } else if (builtin.ident == "probe") {
-    builtin.probe_id = get_probe_id();
-    expr_ = b_.getInt64(builtin.probe_id);
+    auto probe_str = probefull_;
+    probe_str.resize(builtin.type.GetSize() - 1);
+    Constant *const_str = ConstantDataArray::getString(module_->getContext(),
+                                                       probe_str,
+                                                       true);
+    Value *buf = b_.CreateProbeStrAllocation(builtin.type,
+                                             "probe",
+                                             builtin.loc);
+    bool is_alloc = dyn_cast<AllocaInst>(buf);
```

Comment:
LLVM has `isa` for this purpose:
```suggestion
    bool is_alloc = isa<AllocaInst>(buf);
```